analytic_base_department/models/analytic.py

```diff
@@ -0,0 +1,37 @@
+# © 2011-2016 Camptocamp SA
+# © 2013 Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class AnalyticAccount(models.Model):
+    _inherit = "account.analytic.account"
+
+    department_id = fields.Many2one(
+        'hr.department',
+        'Department')
+
+
+class AnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    @api.multi
+    def _get_department(self):
+        department_id = False
+        employee = self.env.user.employee_ids[0]
+        if employee and employee.department_id:
+            department_id = employee.department_id.id
+        return department_id
+
+    department_id = fields.Many2one(
+        'hr.department',
+        'Department',
+        default=_get_department,
```

Comment:
don't reference the method's object. Proxy it w/ lambda: `default=lambda self: self._default_department()`